dev-client/src/components/tables/soilProperties/SoilPropertiesData.tsx

```diff
@@ -43,24 +42,26 @@ export type SoilPropertiesDataTableRow = {
 export const rowsFromSoilData = (
   data: SoilData,
 ): SoilPropertiesDataTableRow[] => {
-  return data.depthDependentData.map(d => rowFromSoilData(d));
+  return data.depthIntervals.map(interval => rowFromSoilData(data, interval));
 };
 
 export const rowFromSoilData = (
-  data: DepthDependentSoilData,
+  data: SoilData,
+  interval: SoilDataDepthInterval,
 ): SoilPropertiesDataTableRow => {
-  const color = fullMunsellColor({
-    colorHue: data.colorHue,
-    colorChroma: data.colorChroma,
-    colorValue: data.colorValue,
-  });
+  const dataForInterval = data.depthDependentData.find(sameDepth(interval));
+
+  const color =
+    dataForInterval &&
+    fullMunsellColor({
+      colorHue: dataForInterval.colorHue,
+      colorChroma: dataForInterval.colorChroma,
+      colorValue: dataForInterval.colorValue,
+    });
   return {
-    depth: {
-      start: data.depthInterval.start,
-      end: data.depthInterval.end,
-    },
-    texture: data.texture ?? undefined,
-    rockFragment: data.rockFragmentVolume ?? undefined,
+    depth: interval.depthInterval,
+    texture: dataForInterval?.texture ?? undefined,
```

Comment:
question: if you don't have the `?? undefined`, do you get some other default value?